### PR TITLE
teku-5930 Make it the default to not sync from Genesis

### LINKED ISSUE #5930: Make it the default to not sync from Genesis
A weird anti-pattern is that a new Teku installation defaults to syncing from Genesis although we strongly recommend against doing this and encourage people to use checkpoint sync instead.

This is really a conversation starter about how we might go about changing this. Easiest may be to simply to "bake in" a recent checkpoint with releases. In a sense this is what we do with the Genesis file, though post-genesis checkpoints are more controversial.  Alternatively, find a set of "trusted" state providers and point `--initial-state` to one of them, perhaps after verifying that they all agree. For the naive user, this get them up and running quickly, and as safely as syncing from Genesis; the sophisticated user can override this to choose their own provider.

There is a wider conversation going on about how to verify checkpoint states and we should be sure to work in concert with that as it evolves.

### PR DESCRIPTION
**Pull request by lucassaldanha**
## PR Description

One common issue when starting a new node is start synchronizing from genesis, instead of using checkpoint sync. Not only this takes longer to sync, but is riskier (subject to long-range attacks).

By default, Teku won't allow users to sync from an epoch that is too old (out of the weak subjectivity period). This includes trying to sync from genesis on mainnet.

For most users, the best option to start a new node is to use [`--checkpoint-sync-url`](https://docs.teku.consensys.net/development/get-started/checkpoint-start) pointing to a checkpoint sync endpoint in sync with the network. This is going to start their node from the latest finalized checkpoint.

This PR added a CLI option `--ignore-weak-subjectivity-period-enabled` to allow users to opt-out of this behaviour.

## Fixed Issue(s)
fixes #5930 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: services/beaconchain/src/test/java/tech/pegasys/teku/services/beaconchain/WeakSubjectivityInitializerTest.java [hunk 1]
```diff
@@ -226,11 +231,68 @@ public void validateInitialAnchor_atMostRecentPotentiallyFinalizedEpoch() {
     final UInt64 currentSlot = spec.computeStartSlotAtEpoch(currentEpoch);
 
     final DataStructureUtil dataStructureUtil = new DataStructureUtil(spec);
-    final UInt64 anchorEpoch = currentEpoch.min(2);
+    final UInt64 anchorEpoch = currentEpoch.minus(2);
```
Comment (lucassaldanha) [inline services/beaconchain/src/test/java/tech/pegasys/teku/services/beaconchain/WeakSubjectivityInitializerTest.java:234]: This was a bug in one of the tests.

File: teku/src/main/java/tech/pegasys/teku/cli/options/Eth2NetworkOptions.java [hunk 2]
```diff
@@ -49,6 +49,16 @@ public class Eth2NetworkOptions {
       arity = "1")
   private String initialState;
 
+  @Option(
+      names = {"--ignore-weak-subjectivity-period-enabled"},
+      paramLabel = "<BOOLEAN>",
+      description = "Foo.",
```
Comment (rolfyone) [inline teku/src/main/java/tech/pegasys/teku/cli/options/Eth2NetworkOptions.java]: could arguably improve the description...

File: teku/src/main/java/tech/pegasys/teku/cli/options/Eth2NetworkOptions.java [hunk 3]
```diff
@@ -49,6 +49,16 @@ public class Eth2NetworkOptions {
...
+      arity = "0..1",
+      fallbackValue = "true",
+      defaultValue = "false",
```
Comment (rolfyone) [inline teku/src/main/java/tech/pegasys/teku/cli/options/Eth2NetworkOptions.java]: fairly sure we dont generally set the default value in a boolean, just a fallback, and the boolean default is in the variable initialisation...

File: services/beaconchain/src/test/java/tech/pegasys/teku/services/beaconchain/WeakSubjectivityInitializerTest.java [hunk 4]
```diff
@@ -226,11 +231,68 @@ public void validateInitialAnchor_atMostRecentPotentiallyFinalizedEpoch() {
...
+    final WeakSubjectivityCalculator wsCalculator = mock(WeakSubjectivityCalculator.class);
+    doReturn(true).when(wsCalculator).isWithinWeakSubjectivityPeriod(any(), eq(currentSlot));
+
+    // Should not throw
+    initializer.validateInitialAnchor(anchor, currentSlot, spec, Optional.of(wsCalculator));
+  }
+
+  @Test
+  public void validateInitialAnchor_outOfWeakSubjectivityPeriod_WhenConfigIsPresent_ShouldFail() {
+    final UInt64 currentEpoch = UInt64.valueOf(10);
```
Comment (rolfyone) [inline services/beaconchain/src/test/java/tech/pegasys/teku/services/beaconchain/WeakSubjectivityInitializerTest.java:261]: `validateInitialAnchor_withSubjectivityCheck_outOfWeakSubjectivityPeriod_ShouldFail`

seeing if i can make names work....

File: services/beaconchain/src/test/java/tech/pegasys/teku/services/beaconchain/WeakSubjectivityInitializerTest.java [hunk 5]
```diff
@@ -226,11 +231,68 @@ public void validateInitialAnchor_atMostRecentPotentiallyFinalizedEpoch() {
...
         dataStructureUtil.randomAnchorPoint(anchorEpoch, spec.fork(currentEpoch));
 
     // Should not throw
-    initializer.validateInitialAnchor(anchor, currentSlot, spec);
+    initializer.validateInitialAnchor(anchor, currentSlot, spec, Optional.empty());
+  }
+
+  @Test
+  public void
+      validateInitialAnchor_withinWeakSubjectivityPeriod_WhenConfigIsPresent_ShouldSucceed() {
```
Comment (rolfyone) [inline services/beaconchain/src/test/java/tech/pegasys/teku/services/beaconchain/WeakSubjectivityInitializerTest.java]: `validateInitialAnchor_withSubjectivityCheck_withinSubjectivityPeriod_shouldSucceed`

File: services/beaconchain/src/test/java/tech/pegasys/teku/services/beaconchain/WeakSubjectivityInitializerTest.java [hunk 6]
```diff
@@ -226,11 +231,68 @@ public void validateInitialAnchor_atMostRecentPotentiallyFinalizedEpoch() {
...
+            () ->
+                initializer.validateInitialAnchor(
+                    anchor, currentSlot, spec, Optional.of(wsCalculator)))
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessageContaining("Cannot sync outside of weak subjectivity period.");
+  }
+
+  @Test
+  public void
+      validateInitialAnchor_outOfWeakSubjectivityPeriod_WhenConfigIsNotPresent_ShouldSucceed() {
```
Comment (rolfyone) [inline services/beaconchain/src/test/java/tech/pegasys/teku/services/beaconchain/WeakSubjectivityInitializerTest.java]: `validateInitialAnchor_withoutSubjectivityCheck_outOfWeakSubjectivityPeriod_shouldSucceed`

**Review by rolfyone — APPROVED** [review]
mostly nits, looks good.

**Pull request merged**